File: src/gestorAplicacion/registrosMedicos/HistoriaClinica.java
package gestorAplicacion.registrosMedicos;
import gestorAplicacion.personas.Paciente;
import java.util.ArrayList;

public class HistoriaClinica {
    private Paciente paciente;
    public static ArrayList<Diagnostico> diagnosticos;
    public static ArrayList<HistoriaClinica> historia= new ArrayList<HistoriaClinica>();
    static HistoriaClinica pacEncontrado;
    
    //constructor
    public HistoriaClinica(Paciente paciente, ArrayList<Diagnostico> diagnosticos) {
        this.paciente = paciente;
        this.diagnosticos = diagnosticos;
    }
    public HistoriaClinica(){}

    //getter & setter
    public Paciente getPaciente() {
        return paciente;
    }

    public void setPaciente(Paciente paciente) {
        this.paciente = paciente;
    }

    public ArrayList<Diagnostico> getDiagnosticos() {
        return diagnosticos;
    }

    public void setDiagnosticos(ArrayList<Diagnostico> diagnosticos) {
        this.diagnosticos = diagnosticos;
    }
    public static HistoriaClinica getPacEncontrado() {
        return pacEncontrado;
    }
    
    //METODOS

    public String visualizarDiagnostico(Diagnostico diagnosticos, int id) {
        return "El diagnostico es "+ diagnosticos + "El id del diagnostico es "+id;

    }
    public void crearHistoriaClinica(Paciente paciente, ArrayList<Diagnostico> diagnosticos){
        HistoriaClinica historia1 = new HistoriaClinica(paciente, diagnosticos);
        historia.add(historia1);
    }
    
    
    public static int buscarPaciente(int solicitado){// se ingresa el num de la cc y se busca en la lista cuantas hay
        for (HistoriaClinica ee : historia) {
            if(ee.getPaciente().getNumeroDocumento()==solicitado){
                pacEncontrado= ee.getPaciente().consultarHistorioaClinica();
                return 1;
            }
        }
        return 0;
    }
    
}

File: practica2-grupo01-Equipo09/src/gestorAplicacion/registrosMedicos/HistoriaClinica.py
#La clase HistoriaClinica es en donde se puede ver la historia del paciente.
#Esta clase no tiene metodos asociados
from gestorAplicacion.personas.Administrador import Administrador
class HistoriaClinica:
    def __init__(self, paciente, diagnosticos): 
        self._paciente = paciente
        self._diagnosticos = diagnosticos
        Administrador.historiasClinicas.append(self)
    
    #Getters y Setters
    def getPaciente(self):
        return self._paciente

    def setPaciente(self, paciente):
        self._paciente = paciente

    def getDiagnosticos(self):
        return self._diagnosticos

    def setDiagnosticos(self, diagnosticos):
        self._diagnosticos = diagnosticos        
